src/main/java/picard/sam/markduplicates/util/ReadEndsForMarkDuplicates.java

```diff
@@ -62,7 +63,7 @@ public ReadEndsForMarkDuplicates(final ReadEndsForMarkDuplicates read) {
         this.read1Coordinate = read.read1Coordinate;
         this.read2ReferenceIndex = read.read2ReferenceIndex;
         this.read2Coordinate = read.read2Coordinate;
-
+        this.read2CoordinateRequiresSerialize = read.read2CoordinateRequiresSerialize;
```

Comment:
I might rename this to either  `serializeRead2Coordinate` or `read2CoordinateRequiresSerialization` to just be a bit more idiomatic.